BlueprintUI/Sources/Environment/Environment.swift

```diff
@@ -69,8 +69,50 @@ public struct Environment {
         }
     }
     
+    /// Returns a new `Environment` by merging the values from `self` and the
+    /// provided environment; keeping values from the provided environment when there
+    /// are key overlaps between the two environments.
+    func merged(prioritizing other : Environment) -> Environment {
+        var merged = self
+        merged.values.merge(other.values) { $1 }
+        return merged
+    }
+        
     /// Cache used to speed up measurements during a layout pass.
     /// This is declared on the `Environment` directly, because access to the `measurementCache` is a hot path
     /// which we can optimize by not needing the usual `env.values[...]` lookup.
     var measurementCache : MeasurementCache
 }
+
+
+extension UIView {
+    
+    /// The ``Environment`` for the ``Element`` that this view represents in a Blueprint element tree,
+    /// or if the view is not explicitly managed by Blueprint, the ``Environment`` of
+    /// the nearest superview that is managed by Blueprint.
+    ///
+    /// If no views in the superview hierarchy are managed by Blueprint, this property returns nil.
+    var inheritedBlueprintEnvironment : Environment? {
```

Comment:
I kept this internal for now; but would this be helpful at all, eg in the Market view layer; would it simplify anything?